### PR TITLE
Fix for saving many-many related data with ASolrSearchable

### PR DESCRIPTION
**Pull request by bayeru**
I made some changes on ASolrSearchable to get it work with many-many relations. Please review.

### CODE REVIEW

**Comment by phpnode** [comment]
could you post an example field declaration from your solr config so i can see how you're storing this?

**Comment by bayeru** [comment]
Here are the field declarations;

``` xml
<fields>
   <field name="id" type="string" indexed="true" stored="true" required="true" />
   <field name="title" type="text_general" indexed="true" stored="true" />
   <field name="url" type="string" indexed="true" stored="true" />
   <field name="thumbnail_url" type="string" indexed="false" stored="true" />
   <field name="description" type="text_general" indexed="true" stored="true" />
   <field name="create_time" type="date" indexed="true" stored="true" default="NOW/SECOND" />
   <field name="colors__name" type="string" indexed="true" stored="true" multiValued="true" />
   <!-- Red | Blue | Green -->
   <field name="text" type="text_general" indexed="true" stored="false" multiValued="true"/>
 </fields>
```

**Comment by bayeru** [comment]
This is my first pull request ever. I created another pull request for fixing setLimit(0) but the comment appeared here. I don't know why.

File: ASolrQueryResponse.php [hunk 1]
```diff
@@ -1,238 +1,240 @@
-<?php
```
Comment (phpnode) [inline ASolrQueryResponse.php:1]: For some reason this is all shown as changed, I think it's down to different whitespace. Please make sure you're using tabs everywhere.

Comment (bayeru) [inline ASolrQueryResponse.php:1]: I indeed use tabs everywhere.

**Comment by phpnode** [comment]
Thanks for this, due to the weird whitespace issue i wasn't able to merge this directly, but i replicated your commits in:
7e5087c4ae03d5faaf2d48bebc0c244763ad443e and e1aa0737a51ca07200378c60ae02c81b29607e23

**Pull request closed without merging**